### PR TITLE
feat: create sendToFcm util

### PR DESCRIPTION
**Pull request by alexruzenhack**
### Acceptance Criteria
- create sendToFcm util - send push notification to Android
- handle invalid device id - unregister invalid device id

### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -717,7 +717,7 @@ export interface SendNotificationToDevice {
   deviceId: string,
   title: string,
   description: string,
-  metadata: Record<string, unknown>,
+  metadata: Record<string, string>,
```
Comment (tuliomir) [inline src/types.ts:720]: Here we're enforcing we will only have a 1-level-deep object as metadata.

It serves all of our needs, but since we're restricting the contents I believe a simple comment would help future developers identify this limitation to avoid any errors.

File: src/utils/pushnotification.utils.ts [hunk 2]
```diff
@@ -1,5 +1,8 @@
 import { Lambda } from 'aws-sdk';
 import { SendNotificationToDevice } from '@src/types';
+import { credential, initializeApp, messaging, ServiceAccount } from 'firebase-admin';
+import { MulticastMessage } from 'firebase-admin/messaging';
+import serviceAccount from '@src/utils/fcm.config.json';
```
Comment (tuliomir) [inline src/utils/pushnotification.utils.ts]: Since we're adding a new file outside our repository, I think it deserves its own section on the `readme.md` project file.

Also, a verification to check if the file is not available and a custom error / explicit warning on this context would help developers troubleshoot any problems they might encounter because of this feature.

File: src/utils/pushnotification.utils.ts [hunk 3]
```diff
@@ -14,15 +17,40 @@ if (!process.env.STAGE) {
...
+    if (multicastResult.failureCount === 0) {
+      return { success: true };
+    }
+
+    const { 0: { error } } = multicastResult.responses;
+    if (/token-not-registered/.test(error?.code || '')) {
+      return { success: false, errorMessage: PushNotificationError.INVALID_DEVICE_ID };
+    }
+
+    return { success: false, errorMessage: PushNotificationError.UNKNOWN };
```
Comment (tuliomir) [inline src/utils/pushnotification.utils.ts:65]: If we're going to hide the actual message from the rest of the application, I think we should log it here.

**Review by andreabadesso — APPROVED** [review]

File: src/utils/pushnotification.utils.ts [hunk 4]
```diff
@@ -2,21 +2,32 @@ import { Lambda } from 'aws-sdk';
 import { SendNotificationToDevice } from '@src/types';
 import { credential, initializeApp, messaging, ServiceAccount } from 'firebase-admin';
 import { MulticastMessage } from 'firebase-admin/messaging';
-import serviceAccount from '@src/utils/fcm.config.json';
 import createDefaultLogger from '@src/logger';
 
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const serviceAccount = require('@src/utils/fcm.config.json');
```
Comment (tuliomir) [inline src/utils/pushnotification.utils.ts:8]: I understand this is the simplest way to load a JSON into the application, but it also triggers some static code rules.

I think we can keep it this way and later, if necessary, convert this to a full `fsPromises.readFile` + `JSON.parse` call.

File: tests/db.test.ts [hunk 5]
```diff
@@ -2644,11 +2646,217 @@ describe('getPushDevice', () => {
...
+
+    // get settings querying only devices not loaded on database, resulting on empty list
+    const notRegisteredWalletIdList = devicesToNotLoad.map((each) => each.walletId);
+    const result = await getPushDeviceSettingsList(mysql, notRegisteredWalletIdList);
+
+    // assert settings
+    expect(result).toStrictEqual([]);
+  });
+
+  it('should return a list of seetings even when some wallet ids are not found', async () => {
```
Comment (tuliomir) [inline tests/db.test.ts]:
```suggestion
  it('should return a list of settings even when some wallet ids are not found', async () => {
```

File: package.json [hunk 6]
```diff
@@ -7,7 +7,7 @@
     "jest": "jest --runInBand --collectCoverage --detectOpenHandles --forceExit",
     "lint": "eslint src/ tests/ --ext .js,.jsx,.ts,.tsx src tests",
     "lint-fix": "eslint src/ tests/ --fix --ext .js,.jsx,.ts,.tsx src tests",
-    "check-types": "tsc --noemit",
+    "check-types": "tsc --noemit --skipLibCheck",
```
Comment (tuliomir) [inline package.json:10]: Why was that necessary?

I was expecting it to have some relation to the `bitcore-lib` that we usually have conflicts with, but found no issue with it.

Comment (alexruzenhack) [reply package.json:10]: It will avoid running type checks on the libs in the node_modules path.

**Review by tuliomir — APPROVED** [review]
